### ISSUE TITLE
StatusEffect -> MobEffect/CreatureEffect/Effect

### ISSUE THREAD

**Issue by Runemoro** [asiekierka/MinecraftPOMF#584]
The "Status" prefix adds no useful information to the class name. All it tells you is that it affects the "status" of some thing (which is what any effect does, by definition).

I suggest changing it to either:
 - `MobEffect` (it's slightly wrong since players are probably not considered mobs, but it's the Mojang name)
 - `CreatureEffect`
 - `Effect`

**Comment by liach** [comment]
This was a name from the Minecraft wiki. I will go ask why "Status Effect".

**Comment by Prospector** [comment]
It's an effect that tells the status of an entity. Seems better than MobEffect or CreatureEffect to me. Effect is just very vague.

**Comment by Prospector** [comment]
#198

**Comment by Runemoro** [comment]
> It's an effect that tells the status of an entity. Seems better than MobEffect or CreatureEffect to me. Effect is just very vague.

I agree that Effect is very vague. But adding Status to it doesn't make it less vague. It doesn't mention it's the status *of an entity*, which is why I'm suggesting MobEffect/CreatureEffect/EntityEffect (or MobStatusEffect/CreatureStatusEffect/EntityStatusEffect if you prefer).

Mob is the Mojang name, which is why I'd vote for that, but Creature (or LivingEntity, but that's long) would be more precise since it applies to living entities.

**Comment by Prospector** [comment]
I'm ok with MobEffect if we rename LivingEntity to Mob as mojang has.

**Comment by Runemoro** [comment]
> if we rename LivingEntity to Mob as mojang has.

Mojang only calls non-player living entities mobs. I remember seeing an `instanceof MobEntity` right before an exception message with the word "mob" in it, so I think our current MobEntity class matches the Mojang name (apart from the Entity suffix).

**Comment by Pyrofab** [comment]
I don't really see the point. Status effect is a common name for it, so I doubt anyone would be really confused. `Effect` is too short to mean much, `MobEffect` is slightly wrong, `CreatureEffect` doesn't correspond to our naming of entities.

**Comment by Runemoro** [comment]
> Effect is too short to mean much

And how does adding Status help? By the definition of the word "effect", *all* effects modify some status. It would be like naming it `ModifyingEffect`.

Also, it's not true it doesn't mean anything. The command is named `/effect` and it's clear what it does. We have even shorter names like `Item` which on their own don't mean anything, yet are very clear.

> Status effect is a common name for it

I've only heard players call it "effect". If you're saying it's a common name, then you mean it's a name commonly used by modders because of the MCP name, which we shouldn't care about, the same way we don't care about `TileEntity`.

**Comment by Pyrofab** [comment]
StatusEffect is definitely not an MCP name. In fact I am glad to see that the MCP name hasn't been brought up once in this discussion (because it's not a good one). I am however referring to the official minecraft wiki, which names effects like that: https://minecraft.gamepedia.com/Status_effect. Given that a lot of players get their information from the wiki, I think it is in fact fair to say that it is common.

**Comment by Prospector** [comment]
well...whatever. I'm okay with MobEffect I guess just to go with the mojang name. But I also believe in the Menu change to align with mojang
